eng/Directory.Build.Common.targets

```diff
@@ -98,6 +98,15 @@
 
   </ItemGroup>
 
+  <!-- Remove packages built into the .NET 6+ runtime -->
+  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0' OR '$(TargetFramework)' == 'net8.0'">
```

Comment:
That's interesting.  Maybe we do an explicit `netstandard2.0` check with the compatible flag?

```xml
<ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0' OR '$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0'))' == 'false'">
```

Or, maybe we should use `netstandard2.0` and `net462` as our hardcodes and move to a negative check.   Those will be slower to change over time and result in fewer updates needed.     

```xml
<ItemGroup Condition="'$(TargetFramework)' != 'netstandard2.0' AND '$(TargetFramework)' != 'net462'">
```

If we go down this path, it may be a good idea to define an MS Build variable for "$netFrameworkCompatTarget" somewhere and use that here and in Azure.Core/others where we have an explicit .NET Framework fallback.

I wonder if we should throw an email out to Immo and Eric and see if they have a suggestion for how to do this in a way that we wouldn't have to update over time.

Reply:
I sent an email and will follow up. If there isn't a better way I will change this to be a negative check.